### DISCUSSION TITLE
Keycloak authentication with supabase

### DISCUSSION THREAD

**Discussion by abeldebruijn** [supabase/supabase#4727] [Questions] [answered]
The goal is to use KeyCloak authentication and combine it with Row Level Security of Supabase ([keycloak](https://www.keycloak.org/)). 

When a user logs in we receive the following information from keycloak:

```javascript
const keycloak = {
    // ...
    "refreshToken": "<REFRESH TOKEN>",
    // ...
    "idToken": "<ID TOKEN>",
    "idTokenParsed": {
        "exp": 1640782436,
        "iat": 1640782136,
        "auth_time": 1640779648,
        "jti": "f4acda00-cf0a-4677-b12f-5b05de1c7a64",
        "iss": "https://keycloak-service-dot-wikitest.ew.r.appspot.com/auth/realms/<HIDDEN>",
        "aud": "supabase",
        "sub": "1aa0601d-f08e-4899-b3fb-36c8abd60657",
        "typ": "ID",
        "azp": "supabase",
        "nonce": "2d2e2f1b-cbb5-4615-a2e6-ffd352154028",
        "session_state": "1a4e235c-5f1c-4d27-a37e-32ee8983bf44",
        "at_hash": "NdjchvDSEHgccMtxldTDnQ",
        "acr": "0",
        "sid": "1a4e235c-5f1c-4d27-a37e-32ee8983bf44",
        "email_verified": false,
        "name": "user name",
        "preferred_username": "user@hotmail.com",
        "given_name": "user",
        "family_name": "name",
        "email": "user@hotmail.com"
    },
    "token": "<TOKEN>",
    // ...
}
```

For a demo of how to get this, follow this guide: https://medium.com/keycloak/secure-vue-js-app-with-keycloak-94814181e344

We tried to follow this [this supabase discussion page](https://github.com/supabase/supabase/discussions/417), for custom auth providers. However we this discussion page is not really going in depth in the custom auth providers like keycloak. We hope this discussion page can give more info into custom auth providers

**Comment by magick93** [comment]
This would be great to have an integration with keycloak. Did you manage to make this work?

**Comment by GaryAustin1** [comment, marked as answer]
Keycloak support was announced this week.

https://supabase.com/docs/guides/auth/auth-keycloak

**Reply by magick93** [reply]
Awesome! Thanks!

**Reply by cemremengu** [reply]
Fantastic now we can switch!
